pkgs/applications/blockchains/zcash/default.nix

```
@@ -1,38 +1,102 @@
-{ autoreconfHook, boost180, cargo, coreutils, curl, cxx-rs, db62, fetchFromGitHub
-, git, hexdump, lib, libevent, libsodium, makeWrapper, rust, rustPlatform
-, pkg-config, Security, stdenv, testers, tl-expected, utf8cpp, util-linux, zcash, zeromq
+{ autoreconfHook
+, boost
+, coreutils
+, curl
+, cxx-rs
+, db
+, fetchFromGitHub
+, git
+, hexdump
+, lib
+, libevent
+, libsodium
+, llvmPackages
+, makeWrapper
+, overrideCC
+, pkg-config
+, rust
+, rustPlatform
+, Security
+, stdenv
+, testers
+, tl-expected
+, utf8cpp
+, util-linux
+, zcash
+, zeromq
 }:
 
-rustPlatform.buildRustPackage.override { inherit stdenv; } rec {
+let
+  cpu = stdenv.targetPlatform.parsed.cpu.name;
+
+  clangStdenv = if stdenv.isDarwin
+    then llvmPackages.libcxxStdenv
+    else overrideCC stdenv (llvmPackages.libcxxClang.override (old: {
+      bintools = llvmPackages.bintools;
+
+      nixSupport.cc-cflags = (old.nixSupport.cc-cflags or []) ++ [
+        "-rtlib=compiler-rt"
+        "-Wno-unused-command-line-argument"
+      ];
+
+      nixSupport.cc-ldflags = (old.nixSupport.cc-ldflags or []) ++ [
+        "${llvmPackages.compiler-rt}/lib/linux/libclang_rt.builtins-${cpu}.a"
+      ];
+
+      nixSupport.libcxx-cxxflags = (old.nixSupport.libcxx-cxxflags or []) ++ [
+        "-unwindlib=libunwind"
+      ];
+
+      # https://github.com/NixOS/nixpkgs/issues/201591
+      nixSupport.libcxx-ldflags = (old.nixSupport.libcxx-ldflags or []) ++ [
+        "-L${llvmPackages.libunwind}/lib"
+        "-lunwind"
+      ];
+    }));
+
+  boost' = boost.override { stdenv = clangStdenv; };
+
+  db' = db.override { stdenv = clangStdenv; };
+in
+rustPlatform.buildRustPackage.override { stdenv = clangStdenv; } rec {
   pname = "zcash";
-  version = "5.4.2";
+  version = "5.6.1";
```

This also needs a src.hash and Cargo.lock update.

I'm happy to push a commit that does that, but I wanted to check with you first.

```suggestion
  version = "5.7.0";
```